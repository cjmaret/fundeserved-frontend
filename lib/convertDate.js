export default function convertDate(dateValue) {
  const date = new Date(dateValue.toString());
  let options = {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  };
  return new Intl.DateTimeFormat('en-US', options).format(date);
}
