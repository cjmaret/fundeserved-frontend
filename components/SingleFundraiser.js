import { useMutation, useQuery } from '@apollo/client';
import gql from 'graphql-tag';
import DisplayError from './ErrorMessage';
import Footer from '../components/Footer';
import {
  FundraiserSection,
  TitleGroup,
  Title,
  UpdateButton,
  FundraiserInfo,
  Details,
  Sidebar,
  Image,
  Description,
  AmountGroup,
  AmountValue,
  AmountRaised,
  ButtonGroup,
  ShareButton,
  DonateButton,
  DonorList,
  DonorCard,
  DonorCardPhoto,
  DonorCardName,
  DonorCardAmount,
  DonorListSeeAllButton,
  DonorCardDetails,
  PercentageBarGroup,
  PercentageBar,
  PercentageBarFilled,
  UpdateModal,
  DeleteModal,
  UpdateForm,
  UpdateButtonGroup,
  DeleteFormTitle,
  DeleteFormButton,
  DeleteFormGroup,
  DonorsModal,
  DonateModalList,
  DonorsModalContentGroup,
  DonorsModalTitle,
  DonorsModalList,
  DonorsModalDetails,
  DonorsModalName,
  DonorsModalAmount,
  DonorsModalPhoto,
  DonorsModalCard,
  ShareModal,
  SharePopup,
  Dot,
  DonorsModalDate,
  ModalDot,
  DonorsModalDonationDetails,
  DonorCardDonationDetails,
  DonorCardDate,
} from './styles/styledSingleFundraiser';
import { CloseIcon } from './styles/styledModal';
import { donors } from '../array-data/donors';
import { formatCentsToDollars } from '../lib/formatMoney';
import useForm from '../lib/useForm';
import CloseIconImage from '../images/close-icon.png';
import { useEffect, useState } from 'react';
import Router from 'next/router';
import BlankProfileImage from '../images/blank-profile.jpg';
import CopyIconImage from '../images/copy-icon.png';
import daysAgo from '../lib/daysAgo';
import { useUser } from './User';

export const SINGLE_FUNDRAISER_QUERY = gql`
  query SINGLE_FUNDRAISER_QUERY($id: ID!) {
    Fundraiser(where: { id: $id }) {
      id
      name
      description
      status
      amount
      goal
      user {
        id
      }
      photo {
        id
        altText
        image {
          publicUrlTransformed
        }
      }
      donations {
        total
        dateCreated
        user {
          name
          avatar {
            publicUrlTransformed
          }
        }
      }
    }
  }
`;

const UPDATE_FUNDRAISER_WITH_IMAGE_MUTATION = gql`
  mutation UPDATE_FUNDRAISER_WITH_IMAGE_MUTATION(
    $id: ID!
    $name: String
    $description: String
    $image: Upload
  ) {
    updateFundraiser(
      id: $id
      data: {
        name: $name
        description: $description
        photo: { create: { image: $image, altText: $name } }
      }
    ) {
      id
      name
      description
      photo {
        id
        altText
        image {
          publicUrlTransformed
        }
      }
    }
  }
`;

const UPDATE_FUNDRAISER_NO_IMAGE_MUTATION = gql`
  mutation UPDATE_FUNDRAISER_NO_IMAGE_MUTATION(
    $id: ID!
    $name: String
    $description: String
  ) {
    updateFundraiser(
      id: $id
      data: { name: $name, description: $description }
    ) {
      id
      name
      description
      photo {
        id
        altText
        image {
          publicUrlTransformed
        }
      }
    }
  }
`;

const DELETE_FUNDRAISER_MUTATION = gql`
  mutation DELETE_FUNDRAISER_MUTATION($id: ID!) {
    deleteFundraiser(id: $id) {
      id
    }
  }
`;

const DELETE_FUNDRAISER_IMAGE_MUTATION = gql`
  mutation DELETE_FUNDRAISER_IMAGE_MUTATION($fundraiserImageId: ID!) {
    deleteFundraiserImage(id: $fundraiserImageId) {
      id
    }
  }
`;

function update(cache, payload) {
  console.log(payload);
  cache.evict(cache.identify(payload.data.deleteFundraiser));
}

export default function SingleFundraiser({ id }) {
  const user = useUser();
  const [isUpdateModalOpen, setIsUpdateModalOpen] = useState(false);
  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
  const [isDonorsModalOpen, setIsDonorsModalOpen] = useState(false);
  const [isShareModalOpen, setIsShareModalOpen] = useState(false);

  const { data, loading, error } = useQuery(SINGLE_FUNDRAISER_QUERY, {
    variables: { id },
  });

  const [
    updateFundraiserWithImage,
    {
      data: updateDataWithImage,
      loading: updateLoadingWithImage,
      error: updateErrorWithImage,
    },
  ] = useMutation(UPDATE_FUNDRAISER_WITH_IMAGE_MUTATION, {
    refetchQueries: [
      {
        SINGLE_FUNDRAISER_QUERY,
      },
    ],
  });

  const [
    updateFundraiserNoImage,
    {
      data: updateDataNoImage,
      loading: updateLoadingNoImage,
      error: updateErrorNoImage,
    },
  ] = useMutation(UPDATE_FUNDRAISER_NO_IMAGE_MUTATION, {
    refetchQueries: [
      {
        SINGLE_FUNDRAISER_QUERY,
      },
    ],
  });

  const [
    deleteFundraiser,
    { loading: deleteFundraiserLoading, error: deleteFundraiserError },
  ] = useMutation(DELETE_FUNDRAISER_MUTATION, {
    variables: {
      id,
    },
    update,
  });

  const [
    deleteFundraiserImage,
    {
      loading: deleteFundraiserImageLoading,
      error: deleteFundraiserImageError,
    },
  ] = useMutation(DELETE_FUNDRAISER_IMAGE_MUTATION);

  const Fundraiser = data?.Fundraiser;
  const FundraiserImageSource = Fundraiser?.photo?.image?.publicUrlTransformed;
  const { inputs, handleChange, clearForm, resetForm } = useForm({
    name: Fundraiser?.name || '',
    description: Fundraiser?.description || '',
    image: FundraiserImageSource,
  });

  function checkIfClickedOutside(e) {
    const innerUpdateFormDiv = document.getElementById('update-form');
    const innerDeleteFormDiv = document.getElementById('delete-form');
    const innerDonorsDiv = document.getElementById('donors-div');
    const innerShareDiv = document.getElementById('share-div');
    if (isUpdateModalOpen) {
      if (!innerUpdateFormDiv.contains(e.target)) {
        setIsUpdateModalOpen(false);
      }
    }
    if (isDeleteModalOpen) {
      if (!innerDeleteFormDiv.contains(e.target)) {
        setIsDeleteModalOpen(false);
      }
    }
    if (isDonorsModalOpen) {
      if (!innerDonorsDiv.contains(e.target)) {
        setIsDonorsModalOpen(false);
      }
    }
    if (isShareModalOpen) {
      if (!innerShareDiv.contains(e.target)) {
        setIsShareModalOpen(false);
      }
    }
  }

  async function runImageMutationOrNoImageMutation() {
    if (inputs.image === FundraiserImageSource) {
      await updateFundraiserNoImage({
        variables: {
          id,
          name: inputs.name,
          description: inputs.description,
        },
      })
        .then((res) => {
          setIsUpdateModalOpen(false);
        })
        .catch((err) => console.error(err));
    } else {
      await updateFundraiserWithImage({
        variables: {
          id,
          name: inputs.name,
          description: inputs.description,
          image: inputs.image,
        },
      })
        .then((res) => {
          deleteFundraiserImage({
            variables: {
              fundraiserImageId: Fundraiser.photo.id,
            },
          });
          setIsUpdateModalOpen(false);
        })
        .catch((err) => console.error(err));
    }
  }

  function handleDeleteFundraiser() {
    deleteFundraiser()
      .then((res) => {
        deleteFundraiserImage({
          variables: {
            fundraiserImageId: Fundraiser.photo.id,
          },
        });
        Router.push({
          pathname: `/fundraisers`,
        });
        alert('Fundraiser deleted');
      })
      .catch((err) => console.error(err));
  }

  if (loading) return <p>Loading...</p>;
  if (
    error ||
    updateErrorWithImage ||
    updateErrorNoImage ||
    deleteFundraiserError ||
    deleteFundraiserImageError
  ) {
    return (
      <DisplayError
        error={
          error ||
          updateErrorWithImage ||
          updateErrorNoImage ||
          deleteFundraiserError ||
          deleteFundraiserImageError
        }
      />
    );
  }

  return (
    <>
      <FundraiserSection data-testid="single-fundraiser">
        <TitleGroup>
          {Fundraiser.user?.id === user?.id && (
            <UpdateButtonGroup>
              <UpdateButton onClick={() => setIsUpdateModalOpen(true)}>
                Edit Fundraiser
              </UpdateButton>
              <UpdateButton onClick={() => setIsDeleteModalOpen(true)}>
                Delete Fundraiser
              </UpdateButton>
            </UpdateButtonGroup>
          )}
          <Title>{Fundraiser.name}</Title>
        </TitleGroup>
        <FundraiserInfo>
          <Details>
            <Image src={FundraiserImageSource} alt="" />
            <Description>{Fundraiser.description}</Description>
          </Details>
          <Sidebar>
            <AmountGroup>
              <AmountRaised>
                {formatCentsToDollars(Fundraiser.amount)}{' '}
              </AmountRaised>
              raised of {formatCentsToDollars(Fundraiser.goal)} goal
            </AmountGroup>
            <PercentageBarGroup>
              <PercentageBarFilled
                filled={(Fundraiser.amount / Fundraiser.goal) * 100}
              />
              <PercentageBar />
            </PercentageBarGroup>
            <ButtonGroup>
              <ShareButton onClick={() => setIsShareModalOpen(true)}>
                Share
              </ShareButton>
              <DonateButton href={`/donate/${Fundraiser.id}`}>
                Donate now
              </DonateButton>
            </ButtonGroup>
            <DonorList>
              {Fundraiser.donations.slice(0, 3).map((donor, i) => (
                <DonorCard key={i}>
                  <DonorCardPhoto
                    src={
                      donor.user?.avatar?.publicUrlTransformed ||
                      BlankProfileImage
                    }
                    alt=""
                  />
                  <DonorCardDetails>
                    <DonorCardName>{donor.user?.name}</DonorCardName>
                    <DonorCardDonationDetails>
                      <DonorCardAmount>
                        Donated {formatCentsToDollars(donor.total)}
                      </DonorCardAmount>
                      <DonorCardDate>
                        {daysAgo(donor.dateCreated)} days ago
                      </DonorCardDate>
                    </DonorCardDonationDetails>
                  </DonorCardDetails>
                </DonorCard>
              ))}
            </DonorList>
            <DonorListSeeAllButton onClick={() => setIsDonorsModalOpen(true)}>
              See all
            </DonorListSeeAllButton>
          </Sidebar>
        </FundraiserInfo>
      </FundraiserSection>
      <Footer />
      <UpdateModal
        isUpdateModalOpen={isUpdateModalOpen}
        onClick={checkIfClickedOutside}>
        <UpdateForm
          id="update-form"
          onSubmit={(e) => {
            e.preventDefault();
            runImageMutationOrNoImageMutation();
          }}>
          <CloseIcon
            src={CloseIconImage}
            alt="close icon"
            onClick={() => setIsUpdateModalOpen(false)}
          />
          <DisplayError error={updateErrorNoImage || updateErrorWithImage} />
          <fieldset
            disabled={loading || updateLoadingNoImage || updateLoadingNoImage}
            aria-busy={loading || updateLoadingNoImage || updateLoadingNoImage}>
            <label htmlFor="name">
              Name
              <input
                type="text"
                id="name"
                name="name"
                placeholder="Name"
                value={inputs.name}
                onChange={handleChange}
              />
            </label>
            <label htmlFor="image">
              Image
              <input
                type="file"
                id="image"
                name="image"
                onChange={handleChange}
              />
            </label>
            <label htmlFor="description">
              Description
              <textarea
                type="textarea"
                id="description"
                name="description"
                placeholder="description"
                value={inputs.description}
                onChange={handleChange}
                rows="11"
              />
            </label>
            <div className="button-group">
              <button type="submit">Update Fundraiser</button>
              <button type="button" onClick={clearForm}>
                Clear Form
              </button>
            </div>
          </fieldset>
        </UpdateForm>
      </UpdateModal>
      <DeleteModal
        isDeleteModalOpen={isDeleteModalOpen}
        onClick={checkIfClickedOutside}>
        <DeleteFormGroup id="delete-form">
          <CloseIcon
            src={CloseIconImage}
            alt=""
            onClick={() => setIsDeleteModalOpen(false)}
          />
          <DeleteFormTitle>Delete Fundraiser?</DeleteFormTitle>
          <DeleteFormButton
            onClick={handleDeleteFundraiser}
            aria-busy={deleteFundraiserLoading}
            disabled={deleteFundraiserLoading}>
            Delete
          </DeleteFormButton>
        </DeleteFormGroup>
      </DeleteModal>
      <DonorsModal
        isDonorsModalOpen={isDonorsModalOpen}
        onClick={checkIfClickedOutside}>
        <DonorsModalContentGroup>
          <CloseIcon
            src={CloseIconImage}
            alt=""
            onClick={() => setIsDonorsModalOpen(false)}
          />
          <DonorsModalTitle>All Donors</DonorsModalTitle>
          <DonorsModalList id="donors-div">
            {Fundraiser.donations.map((donor, i) => (
              <DonorsModalCard key={i}>
                <DonorsModalPhoto
                  src={
                    donor.user?.avatar?.publicUrlTransformed ||
                    BlankProfileImage
                  }
                  alt=""
                />
                <DonorsModalDetails>
                  <DonorsModalName>{donor.user?.name}</DonorsModalName>
                  <DonorsModalDonationDetails>
                    <DonorsModalAmount>
                      Donated {formatCentsToDollars(donor.total)}
                    </DonorsModalAmount>
                    <DonorsModalDate>
                      {daysAgo(donor.dateCreated)} days ago
                    </DonorsModalDate>
                  </DonorsModalDonationDetails>
                </DonorsModalDetails>
              </DonorsModalCard>
            ))}
          </DonorsModalList>
        </DonorsModalContentGroup>
      </DonorsModal>
      <ShareModal
        isShareModalOpen={isShareModalOpen}
        onClick={checkIfClickedOutside}>
        <SharePopup id="share-div">
          <CloseIcon
            src={CloseIconImage}
            alt=""
            onClick={() => setIsShareModalOpen(false)}
          />
          <h3>Share this fundraiser!</h3>
          <div>
            <p>{`www.gofundyourself.com/fundraiser/${id}`}</p>
            <button
              onClick={() => {
                navigator.clipboard.writeText(
                  `www.gofundyourself.com/fundraiser/${id}`
                );
              }}>
              <img src={CopyIconImage} alt="" />
            </button>
          </div>
        </SharePopup>
      </ShareModal>
    </>
  );
}
