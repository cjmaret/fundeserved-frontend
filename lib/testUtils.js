import casual from 'casual';
import { PAGINATION_COUNT_QUERY } from '../components/Pagination';

casual.seed(777);

const fakeFundraiser = () => ({
  id: 'abc123',
  name: 'test-fundraiser',
  description: 'fundraiser description',
  status: 'ACTIVE',
  amount: 500,
  goal: 600,
  dateCreated: '2022-08-25T16:00:00.000Z',
  user: {
    id: '123',
  },
  photo: {
    id: 'abc123',
    altText: 'test-fundraiser',
    image: {
      publicUrlTransformed: 'dog.jpg',
    },
  },
  donations: [
    {
      total: 50,
      dateCreated: '2022-09-25T16:00:00.000Z',
      user: {
        name: 'Colin',
        avatar: {
          publicUrlTransformed: 'dog.jpg',
        },
      },
    },
  ],
});

const fakeUser = (overrides) => ({
  __typename: 'User',
  id: '4234',
  name: casual.name,
  email: casual.email,
  createdOn: '2022-05-25T16:00:00.000Z',
  avatar: {
    publicUrlTransformed: 'colin.jpg',
  },
  fundraisers: [
    {
      id: '123',
      name: 'fundraiser1',
      description: 'fundraiser',
      amount: 5000,
      goal: 6000,
      dateCreated: '2022-09-25T16:00:00.000Z',
      photo: {
        image: {
          publicUrlTransformed: 'fundraiser1.jpg',
        },
      },
    },
  ],
  donations: [
    {
      id: '123',
      total: 25,
      dateCreated: '2022-05-26T16:00:00.000Z',
      fundraiser: {
        id: '456',
        name: 'fundraiser2',
        description: 'fundraiser',
        photo: {
          image: {
            publicUrlTransformed: 'fundraiser2.jpg',
          },
        },
      },
    },
  ],
  permissions: ['OVERLORD'],

  ...overrides,
});

const fakeOrderItem = () => ({
  __typename: 'OrderItem',
  id: casual.uuid,
  image: {
    image: `${casual.word}.jpg`,
  },
  name: casual.words(),
  price: 4234,
  quantity: 1,
  description: casual.words(),
});

const fakeOrder = () => ({
  __typename: 'Order',
  id: 'ord123',
  charge: 'ch_123',
  total: 40000,
  items: [fakeOrderItem(), fakeOrderItem()],
  createdAt: '2022-12-11T20:16:13.797Z',
  user: fakeUser(),
});

const fakeCartItem = (overrides) => ({
  __typename: 'CartItem',
  id: 'omg123',
  quantity: 3,
  product: fakeItem(),
  user: fakeUser(),
  ...overrides,
});


class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

function makePaginationMocksFor(length) {
  return [
    {
      request: { query: PAGINATION_COUNT_QUERY },
      result: {
        data: {
          _allFundraisersMeta: {
            count: length,
          },
          itemsConnection: {
            __typename: 'aggregate',
            aggregate: {
              count: length,
              __typename: 'count',
            },
          },
        },
      },
    },
  ];
}

export {
  makePaginationMocksFor,
  LocalStorageMock,
  fakeFundraiser,
  fakeUser,
  fakeCartItem,
  fakeOrder,
  fakeOrderItem,
};
