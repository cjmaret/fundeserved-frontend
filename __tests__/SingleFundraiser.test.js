import { MockedProvider } from '@apollo/react-testing';
import { render, screen } from '@testing-library/react';
import SingleFundraiser, {
  SINGLE_FUNDRAISER_QUERY,
} from '../components/SingleFundraiser';
import { fakeFundraiser } from '../lib/testUtils';

const fundraiser = fakeFundraiser();

const mocks = [
  {
    request: {
      query: SINGLE_FUNDRAISER_QUERY,
      variables: {
        id: '123',
      },
    },
    result: {
      data: {
        Fundraiser: fundraiser,
      },
    },
  },
];

describe('<SingleFundraiser', () => {
  it('renders with proper data', async () => {
    const { container, debug } = render(
      <MockedProvider mocks={mocks}>
        <SingleFundraiser id="123" />
      </MockedProvider>
    );
    await screen.findByTestId('single-fundraiser');
    expect(container).toMatchSnapshot();
  });

  it('errors out when an item is not found', async () => {
    const errorMock = [
      {
        request: {
          query: SINGLE_FUNDRAISER_QUERY,
          variables: {
            id: '123',
          },
        },
        result: {
          errors: [{ message: 'Fundraiser not found!' }],
        },
      },
    ];
    const { container, debug } = render(
      <MockedProvider mocks={errorMock}>
        <SingleFundraiser id="123" />
      </MockedProvider>
    );
    await screen.findByTestId('graphql-error');
    expect(container).toHaveTextContent('Fundraiser not found!');
  });
});
