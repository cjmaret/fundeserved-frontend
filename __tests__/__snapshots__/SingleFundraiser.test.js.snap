// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleFundraiser renders with proper data 1`] = `
<div>
  <section
    class="styledSingleFundraiser__FundraiserSection-sc-1kf0v44-0 fiSzri"
    data-testid="single-fundraiser"
  >
    <div
      class="styledSingleFundraiser__TitleGroup-sc-1kf0v44-1 fyDjWg"
    >
      <h1
        class="styledSingleFundraiser__Title-sc-1kf0v44-2 bRZaIS"
      >
        test-fundraiser
      </h1>
    </div>
    <div
      class="styledSingleFundraiser__FundraiserInfo-sc-1kf0v44-5 ffpKJd"
    >
      <div
        class="styledSingleFundraiser__Details-sc-1kf0v44-6 eBrvyO"
      >
        <img
          alt=""
          class="styledSingleFundraiser__Image-sc-1kf0v44-7 hqyATv"
          src="dog.jpg"
        />
        <p
          class="styledSingleFundraiser__Description-sc-1kf0v44-8 iHahmf"
        >
          fundraiser description
        </p>
      </div>
      <div
        class="styledSingleFundraiser__Sidebar-sc-1kf0v44-9 estGAC"
      >
        <p
          class="styledSingleFundraiser__AmountGroup-sc-1kf0v44-10 fyzxGu"
        >
          <span
            class="styledSingleFundraiser__AmountRaised-sc-1kf0v44-12 xxUbi"
          >
            $5
             
          </span>
          raised of 
          $6
           goal
        </p>
        <div
          class="styledSingleFundraiser__PercentageBarGroup-sc-1kf0v44-13 feXyLQ"
        >
          <div
            class="styledSingleFundraiser__PercentageBarFilled-sc-1kf0v44-15 ivPvKp"
          />
          <div
            class="styledSingleFundraiser__PercentageBar-sc-1kf0v44-14 edWDpo"
          />
        </div>
        <div
          class="styledSingleFundraiser__ButtonGroup-sc-1kf0v44-16 hMoLLM"
        >
          <a
            class="styledSingleFundraiser__Button-sc-1kf0v44-17 styledSingleFundraiser__ShareButton-sc-1kf0v44-18 enLOzC eeJqmn"
          >
            Share
          </a>
          <a
            class="styledSingleFundraiser__Button-sc-1kf0v44-17 styledSingleFundraiser__DonateButton-sc-1kf0v44-19 enLOzC"
            href="/donate/abc123"
          >
            Donate now
          </a>
        </div>
        <div
          class="styledSingleFundraiser__DonorList-sc-1kf0v44-20 jAeREC"
        >
          <div
            class="styledSingleFundraiser__DonorCard-sc-1kf0v44-21 iNSKhh"
          >
            <img
              alt=""
              class="styledSingleFundraiser__DonorCardPhoto-sc-1kf0v44-22 hmQezd"
              src="dog.jpg"
            />
            <div
              class="styledSingleFundraiser__DonorCardDetails-sc-1kf0v44-23 jSQEnr"
            >
              <h3
                class="styledSingleFundraiser__DonorCardName-sc-1kf0v44-24 iWuQGw"
              >
                Colin
              </h3>
              <div
                class="styledSingleFundraiser__DonorCardDonationDetails-sc-1kf0v44-25 iaQEVo"
              >
                <p
                  class="styledSingleFundraiser__DonorCardAmount-sc-1kf0v44-26 eLEVWx"
                >
                  Donated 
                  $0.50
                </p>
                <p
                  class="styledSingleFundraiser__DonorCardAmount-sc-1kf0v44-26 styledSingleFundraiser__DonorCardDate-sc-1kf0v44-27 eLEVWx"
                >
                  16
                   days ago
                </p>
              </div>
            </div>
          </div>
        </div>
        <a
          class="styledSingleFundraiser__Button-sc-1kf0v44-17 styledSingleFundraiser__DonorListSeeAllButton-sc-1kf0v44-28 enLOzC cVFrHB"
        >
          See all
        </a>
      </div>
    </div>
  </section>
  <footer
    class="styledFooter__FooterComponent-sc-1e1bsv4-0 jyrief"
  >
    <div
      class="styledFooter__Logo-sc-1e1bsv4-2 fXPONK"
    >
      <a
        class="styledFooter__NavLink-sc-1e1bsv4-1 bBbVVU logo__anchor"
      >
        <img
          alt=""
          class="footer-logo-image"
          src="[object Object]"
        />
      </a>
    </div>
    <div
      class="styledFooter__ListGroup-sc-1e1bsv4-3 eLUlUf"
    >
      <ul
        class="styledFooter__List-sc-1e1bsv4-4 styledFooter__FundraiseList-sc-1e1bsv4-6 gYgrfI"
      >
        <h4
          class="styledFooter__ListTitle-sc-1e1bsv4-5 iCvyLs"
        >
          Fundraise for
        </h4>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/"
          >
            My Benefit
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/"
          >
            Someone Else's Loss
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/"
          >
            Fraud/Deceit
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/"
          >
            Mean Spirited Prank
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/"
          >
            A Good Cause (As If, Right?)
          </a>
        </li>
      </ul>
      <ul
        class="styledFooter__List-sc-1e1bsv4-4 styledFooter__LearnMoreList-sc-1e1bsv4-7 gYgrfI"
      >
        <h4
          class="styledFooter__ListTitle-sc-1e1bsv4-5 iCvyLs"
        >
          Learn more
        </h4>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/learn-more"
          >
            How GoFundYourself works
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/learn-more"
          >
            Common questions
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/learn-more"
          >
            Success stories
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/learn-more"
          >
            Supported countries
          </a>
        </li>
      </ul>
      <ul
        class="styledFooter__List-sc-1e1bsv4-4 styledFooter__ResourcesList-sc-1e1bsv4-8 gYgrfI"
      >
        <h4
          class="styledFooter__ListTitle-sc-1e1bsv4-5 iCvyLs"
        >
          Resources
        </h4>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/learn-more"
          >
            Help center
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/learn-more"
          >
            Blog
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/learn-more"
          >
            Press center
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/learn-more"
          >
            Careers
          </a>
        </li>
        <li
          class="styledFooter__ListItem-sc-1e1bsv4-9 gxZrWY"
        >
          <a
            class="styledFooter__ListLink-sc-1e1bsv4-10 dGPkaC"
            href="/learn-more"
          >
            About
          </a>
        </li>
      </ul>
    </div>
    <p
      class="styledFooter__RightsReserved-sc-1e1bsv4-11"
    >
      2022 All Rights Reserved
    </p>
  </footer>
  <div
    class="styledModal__Modal-sc-1l007rp-0 styledSingleFundraiser__UpdateModal-sc-1kf0v44-29 thicZ eTgUEy"
  >
    <form
      class="styledForm__Form-sc-174guj4-0 styledForm__ModalForm-sc-174guj4-4 styledSingleFundraiser__UpdateForm-sc-1kf0v44-30 eWUBJi bgZiUv"
      id="update-form"
    >
      <img
        alt="close icon"
        class="styledModal__CloseIcon-sc-1l007rp-1 cYbGot"
        src="[object Object]"
      />
      <fieldset
        aria-busy="false"
      >
        <label
          for="name"
        >
          Name
          <input
            id="name"
            name="name"
            placeholder="Name"
            type="text"
            value="test-fundraiser"
          />
        </label>
        <label
          for="image"
        >
          Image
          <input
            id="image"
            name="image"
            type="file"
          />
        </label>
        <label
          for="description"
        >
          Description
          <textarea
            id="description"
            name="description"
            placeholder="description"
            rows="11"
            type="textarea"
          >
            fundraiser description
          </textarea>
        </label>
        <div
          class="button-group"
        >
          <button
            type="submit"
          >
            Update Fundraiser
          </button>
          <button
            type="button"
          >
            Clear Form
          </button>
        </div>
      </fieldset>
    </form>
  </div>
  <div
    class="styledModal__Modal-sc-1l007rp-0 styledSingleFundraiser__DeleteModal-sc-1kf0v44-31 thicZ cBMYYP"
  >
    <div
      class="styledSingleFundraiser__DeleteFormGroup-sc-1kf0v44-32 ctmzIK"
      id="delete-form"
    >
      <img
        alt=""
        class="styledModal__CloseIcon-sc-1l007rp-1 cYbGot"
        src="[object Object]"
      />
      <h2
        class="styledSingleFundraiser__DeleteFormTitle-sc-1kf0v44-33"
      >
        Delete Fundraiser?
      </h2>
      <button
        aria-busy="false"
        class="styledSingleFundraiser__DeleteFormButton-sc-1kf0v44-34 cfeeAz"
      >
        Delete
      </button>
    </div>
  </div>
  <div
    class="styledModal__Modal-sc-1l007rp-0 styledSingleFundraiser__DonorsModal-sc-1kf0v44-35 thicZ itYXt"
  >
    <div
      class="styledSingleFundraiser__DonorsModalContentGroup-sc-1kf0v44-36 ffdInM"
    >
      <img
        alt=""
        class="styledModal__CloseIcon-sc-1l007rp-1 cYbGot"
        src="[object Object]"
      />
      <h3
        class="styledSingleFundraiser__DonorsModalTitle-sc-1kf0v44-37 izOrtq"
      >
        All Donors
      </h3>
      <div
        class="styledSingleFundraiser__DonorsModalList-sc-1kf0v44-38 jzUTjj"
        id="donors-div"
      >
        <div
          class="styledSingleFundraiser__DonorCard-sc-1kf0v44-21 styledSingleFundraiser__DonorsModalCard-sc-1kf0v44-39 iNSKhh hZfEeK"
        >
          <img
            alt=""
            class="styledSingleFundraiser__DonorCardPhoto-sc-1kf0v44-22 styledSingleFundraiser__DonorsModalPhoto-sc-1kf0v44-40 hmQezd dmZaxY"
            src="dog.jpg"
          />
          <div
            class="styledSingleFundraiser__DonorCardDetails-sc-1kf0v44-23 styledSingleFundraiser__DonorsModalDetails-sc-1kf0v44-41 jSQEnr btIObx"
          >
            <h3
              class="styledSingleFundraiser__DonorCardName-sc-1kf0v44-24 styledSingleFundraiser__DonorsModalName-sc-1kf0v44-42 iWuQGw MYflW"
            >
              Colin
            </h3>
            <div
              class="styledSingleFundraiser__DonorCardDonationDetails-sc-1kf0v44-25 styledSingleFundraiser__DonorsModalDonationDetails-sc-1kf0v44-43 iaQEVo"
            >
              <p
                class="styledSingleFundraiser__DonorCardAmount-sc-1kf0v44-26 styledSingleFundraiser__DonorsModalAmount-sc-1kf0v44-44 eLEVWx gnoGTj"
              >
                Donated 
                $0.50
              </p>
              <p
                class="styledSingleFundraiser__DonorCardAmount-sc-1kf0v44-26 styledSingleFundraiser__DonorCardDate-sc-1kf0v44-27 styledSingleFundraiser__DonorsModalDate-sc-1kf0v44-45 eLEVWx"
              >
                16
                 days ago
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="styledModal__Modal-sc-1l007rp-0 styledSingleFundraiser__ShareModal-sc-1kf0v44-46 thicZ kUAetP"
  >
    <div
      class="styledSingleFundraiser__SharePopup-sc-1kf0v44-47 xhjwx"
      id="share-div"
    >
      <img
        alt=""
        class="styledModal__CloseIcon-sc-1l007rp-1 cYbGot"
        src="[object Object]"
      />
      <h3>
        Share this fundraiser!
      </h3>
      <div>
        <p>
          www.gofundyourself.com/fundraiser/123
        </p>
        <button>
          <img
            alt=""
            src="[object Object]"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
